### PR TITLE
Add support for CUDA 11.2 by upgrading Torch/Tensorflow/CUDA compatibility maps

### PR DESCRIPTION
**Pull request by andreasjansson**
Also includes fixes for torch version scraper, since the torch json was changed.

### CODE REVIEW

**Comment by bfirsh** [comment]
See also #183 and #228. Did you manage to fix the issues mentioned in those PRs, causing the tests to fail?

Related to #181 
Closes #202 
Closes #204

**Comment by andreasjansson** [comment]
> See also #183 and #228. Did you manage to fix the issues mentioned in those PRs, causing the tests to fail?

Yes, the issue there was that TF names the versions "11.2" and the CUDA base images use the full "11.2.0" notion. Using `version.Equal` instead of plain `==` fixes that.

File: pkg/dockerfile/generator_test.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ predict: predict.py:Predictor
 	require.NoError(t, err)
 
 	expected := `# syntax = docker/dockerfile:1.2
-FROM nvidia/cuda:11.0-cudnn8-devel-ubuntu16.04
+FROM nvidia/cuda:11.2.0-cudnn8-devel-ubuntu20.04
```
Comment (bfirsh) [inline pkg/dockerfile/generator_test.go:95]: I would expect the default Ubuntu and CUDA version to remain stable. What's going on here? Is it non-deterministically picking the Ubuntu version somehow?

**Review by bfirsh — APPROVED** [review]
Happy to get this in as a bodge, but perhaps we need to open another issue about default Ubuntu versions not remaining stable.

File: pkg/dockerfile/generator_test.go (hunk 1 above)
Comment (andreasjansson) [reply pkg/dockerfile/generator_test.go:95]: If you set `gpu: true` without specifying `cuda:` or including any python packages that pick a CUDA version for you, it defaults to the latest CUDA version. NVIDIA's base image uses Ubuntu 20.04 for CUDA 11.2.

Comment (bfirsh) [reply pkg/dockerfile/generator_test.go:95]: Gotchya. I have created an issue here: https://github.com/replicate/cog/issues/330

**Pull request merged**